### PR TITLE
#1096 - created withCommitPropertyLike method on QueryBuilder

### PR DESCRIPTION
**Pull request by lucas-prestes**
created withCommitPropertyLike method on QueryBuilder, made implementations for InMemoryRepository,MongoRepository and SnapshotQuery, and added unit tests

issue #1096

### CODE REVIEW

**Comment by bartoszwalacik** [comment]
thnx, please wait for the CR

**Comment by lucas-prestes** [comment]
not meaning to be rude or to hurry you, as I know very well you must have lots of other things to attend to, but I'm still waiting for the CR

File: javers-core/src/main/java/org/javers/repository/jql/QueryBuilder.java [hunk 1]
```diff
@@ -482,6 +482,19 @@ public QueryBuilder withCommitProperty(String name, String value) {
         return this;
     }
 
+    /**
+     * Only snapshots with a given commit property partially containing a String value.
+     <br/><br/>
```
Comment (bartoszwalacik) [inline javers-core/src/main/java/org/javers/repository/jql/QueryBuilder.java:487]: please give an example here (using SQL syntax):
`property_value LIKE "%"+value+"%"`

**Comment by bartoszwalacik** [comment]
Hi @lucas-prestes , sorry for the delay, your PR looks very good, I have put only one comment concerning javadocs.

File: javers-core/src/test/groovy/org/javers/core/JaversRepositoryE2ETest.groovy [hunk 2]
```diff
@@ -1345,4 +1345,53 @@ class JaversRepositoryE2ETest extends Specification {
             assert javers.findSnapshots(QueryBuilder.byInstanceId(it, SnapshotEntity).build()).size() == 1
         }
     }
-}
\ No newline at end of file
+
+    def "should query for commit property starting with partial text"() {
```
Comment (bartoszwalacik) [inline javers-core/src/test/groovy/org/javers/core/JaversRepositoryE2ETest.groovy:1349]: these 3 tests are very similar, all 3 cases could be tested in a single data test with `where:` and `@Unroll`

**Pull request merged**